index.js

```diff
@@ -159,13 +159,13 @@ class DatGateway extends DatLibrarian {
           return DatLibrarian.resolve(address).then((resolvedAddress) => {
             // TODO: Detect DatDNS addresses
             let encodedAddress = hexTo32.encode(resolvedAddress)
-            let redirectURL = `http://${encodedAddress}.${urlParts.hostname}:${this.server.address().port}/${path}${urlParts.search || ''}`
+            let redirectURL = `http://${encodedAddress}.${urlParts.host}/${path}${urlParts.search || ''}`
 
             log('Redirecting %s to %s', address, redirectURL)
             res.setHeader('Location', redirectURL)
             res.writeHead(302)
             res.end()
-          })
+          }).catch((e) => log(e))
```

Comment:
Instead of logging the error, could you use write a 500 status code to the response?